scripts/accounts-invariants/pallet-balances-maintenance.py

```diff
@@ -333,6 +356,245 @@ def upgrade_accounts(chain_connection,
+        f"Parent state: {parent_account_info_state} " \
+        f"Set storage state: {set_storage_info_state}"
+
+
+def get_system_account_metadata_scale_codec_type(chain_connection):
+    """
+     Retrieves System.Account metadata decoder type
+    :param chain_connection WS connection handler
+    :return: string representation of System.Account decoder type
+    """
+    system_account_storage_function = \
+        next(filter(
+            lambda storage_function: storage_function['storage_name'] == 'Account' and
+                                     storage_function['module_id'] == 'System',
+            chain_connection.get_metadata_storage_functions()))
+    internal_metadata_scale_codec_type = system_account_storage_function['type_value']
+    return internal_metadata_scale_codec_type
+
+
+def fix_double_providers_count_for_account(chain_connection,
+                                           account_id,
+                                           sudo_sender_keypair,
+                                           internal_system_account_scale_codec_type,
+                                           input_args):
+    """
+    Fixes double providers counter for a given account id.
+    :param chain_connection: WS connection handler
+    :param account_id: Account id to which we should fix providers counter
+    :param sudo_sender_keypair Mnemonic phrase of sudo
+    :param internal_system_account_scale_codec_type Internal metadata SCALE decoder/encoder type for System.Account
+           entry
+    """
+    log.info(f"Fixing double providers counter for account {account_id}")
+    log.info(f"Querying state for account {account_id}")
+    result = chain_connection.query('System', 'Account', [account_id])
+    log.debug(f"Returned value: {result.value}")
+    account_id_and_account_info_data = [(account_id, result.value)]
+    raw_key_values = get_raw_key_values(chain_connection,
+                                        account_id_and_account_info_data,
+                                        internal_system_account_scale_codec_type)
+
+    set_storage_call = chain_connection.compose_call(
+        call_module='System',
+        call_function='set_storage',
+        call_params={
+            'items': raw_key_values,
+        })
+    # ref time is set to 400ms to make sure this is the only tx that ends up in a block
+    # 359 875 586 000 is a maximal weight (found empirically) that sudo_unchecked_weight is able to consume
+    max_weight = 359875586000
+    sudo_unchecked_weight_call = chain_connection.compose_call(
+        call_module='Sudo',
+        call_function='sudo_unchecked_weight',
+        call_params={
+            'call': set_storage_call,
+            'weight': {
+                'proof_size': 0,
+                'ref_time': max_weight,
+            },
+        }
+    )
+
+    extrinsic = chain_connection.create_signed_extrinsic(call=sudo_unchecked_weight_call,
```

Comment:
can you add a small tip to the extrinsic, like `0.001` TZERO, to make sure it's first in block?

Reply:
done